tests/fixtures/mocked_panda.py

```diff
@@ -0,0 +1,479 @@
+    """  # noqa: E501
+
+    @contextmanager
+    def ctx() -> Generator[None, None, None]:
+        ctx = get_multiprocessing_context()
+        logger_queue = ctx.Queue()
+        logger = logging.getLogger()
+        logger.addHandler(handlers.QueueHandler(logger_queue))
+        yield
+        while not logger_queue.empty():
+            log_record: logging.LogRecord = logger_queue.get()
+            # Make mypy happy
+            assert (
+                log_record.args
+            ), f"args were none, how did that happen?\nRecord: {log_record}\n"
+            f"Args: {log_record.args}"
+            logger._log(
+                level=log_record.levelno,
+                msg=log_record.message,
+                args=log_record.args,
+                exc_info=log_record.exc_info,
+            )
+
+    return ctx
+
+
+def create_subprocess_ioc_and_responses(
+    response_handler: ResponseHandler,
+    tmp_path: Path,
+    caplog,
+    caplog_workaround,
+    table_field_info,
+    table_fields,
+) -> Generator[Tuple[Path, Connection, ResponseHandler], None, None]:
+    """Run the IOC in its own subprocess. When finished check logging logged no
+    messages of WARNING or higher level."""
+
+    with caplog.at_level(logging.WARNING):
+        with caplog_workaround():
+            ctx = get_multiprocessing_context()
+            parent_conn, child_conn = ctx.Pipe()
+            p = ctx.Process(
+                target=ioc_wrapper,
+                args=(
+                    response_handler,
+                    tmp_path,
+                    child_conn,
+                    table_fields,
+                    table_field_info,
+                    TEST_PREFIX,
+                ),
+            )
+            try:
+                p.start()
+                select_and_recv(parent_conn)  # Wait for IOC to start up
+                yield tmp_path, child_conn, response_handler
+            finally:
+                p.terminate()
+                p.join(10)
+                # Should never take anywhere near 10 seconds to terminate, it's just
+                # there to ensure the test doesn't hang indefinitely during cleanup
+
+    # We expect all tests to pass without warnings (or worse) logged.
+    assert (
+        len(caplog.messages) == 0
+    ), f"At least one warning/error/exception logged during test: {caplog.records}"
+
+
+def Changes_iterator_wrapper(values=None, multiline_values=None):
+    multiline_values = multiline_values or {}
+    return [
+        Changes(
+            values=values, no_value=[], in_error=[], multiline_values=multiline_values
+        ),
+    ]
+
+
+def respond_with_no_changes(number_of_iterations: Optional[int] = None) -> repeat:
+    changes = Changes(
+        values={},
+        no_value=[],
+        in_error=[],
+        multiline_values={},
+    )
+    if number_of_iterations:
+        return repeat(changes, number_of_iterations)
+
+    return repeat(changes)
```

Comment:
Pedantic, but isn't it unecessary to have the `if` statement? According to [the docs](https://docs.python.org/3/library/itertools.html#itertools.repeat) it looks like passing None is the same as not specifying the number of iterations at all.

Reply:
Mypy error, the second argument is expecting an int, so an Optional[int] isn't allowed, I could cast it.

Reply:
Think it should work with 0 being unlimited